ios/Classes/dropIn/DropInPlatformApi.swift

```diff
@@ -283,3 +342,40 @@ extension DropInPlatformApi: DropInInteractorDelegate {
         }
     }
 }
+
+extension DropInPlatformApi: PartialPaymentDelegate {
+    func checkBalance(with data: Adyen.PaymentComponentData, component: any Adyen.Component, completion: @escaping (Result<Adyen.Balance, any Error>) -> Void) {
+        do {
+            checkBalanceHandler = completion
+            let platformCommunicationModel = try PlatformCommunicationModel(
+                type: PlatformCommunicationType.balanceCheck,
+                data: data.jsonObject.toJsonStringRepresentation()
+            )
+            dropInFlutterApi.onDropInAdvancedPlatformCommunication(platformCommunicationModel: platformCommunicationModel, completion: { _ in })
```

Comment:
This has empty completion, it's a good sign to think if it should be optional in the `onDropInAdvancedPlatformCommunication` method.

Reply:
The method `onDropInAdvancedPlatformCommunication` is being generated by the Flutter tool to establish the communication between the cross-platform and native layer. Unfortunately, the code generator doesn't support setting the completion as optional.